src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+
+namespace Bit.Client.Web.BlazorUI
+{
+    public partial class BitProgressIndicator
+    {
+        private bool PercentCompleteHasBeenSet;
+        private bool DescriptionHasBeenSet;
+        private double percentComplete;
+        private string description = string.Empty;
+        [Parameter] public string Label { get; set; } = string.Empty;
+
+        [Parameter]
+        public double PercentComplete
+        {
+            get => percentComplete;
+            set
+            {
+                if (value == percentComplete) return;
+                percentComplete = Normalize(value);
+                _ = PercentCompleteChanged.InvokeAsync(value);
+            }
+        }
+
+        [Parameter]
+        public string Description
+        {
+            get => description;
+            set
+            {
+                if (value == description) return;
+                description = value;
+                _ = DescriptionChanged.InvokeAsync(value);
+            }
+        }
+
+        [Parameter] public EventCallback<double> PercentCompleteChanged { get; set; }
+        [Parameter] public EventCallback<string> DescriptionChanged { get; set; }
+        protected override string RootElementClass => "bit-pi";
+        protected override void RegisterComponentClasses()
+        {
+            ClassBuilder.Register(() => PercentCompleteChanged.HasDelegate
```

Comment:
this is not the correct way to determine the Indeterminate state.
based on the FluentUI library, the Indeterminate state occurs when there is no value set for the PercentComplete parameter.
please find the correct solution for this problem.